src/ui/public/courier/search_strategy/search_strategy_registry.js

```diff
@@ -0,0 +1,69 @@
+/*
+ * Licensed to Elasticsearch B.V. under one or more contributor
+ * license agreements. See the NOTICE file distributed with
+ * this work for additional information regarding copyright
+ * ownership. Elasticsearch B.V. licenses this file to you under
+ * the Apache License, Version 2.0 (the "License"); you may
+ * not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+const searchStrategies = [];
+
+const addSearchStrategy = searchStrategy => {
+  searchStrategies.push(searchStrategy);
+};
+
+/**
+ * Build a structure like this:
+ *
+ *   [{
+ *     searchStrategy: <rollupSearchStrategy>,
+ *     searchRequests: []<SearchRequest>,
+ *   }, {
+ *     searchStrategy: <defaultSearchStrategy>,
+ *     searchRequests: []<SearchRequest>,
+ *   }]
+ *
+ * We use an array of objects to preserve the order of the search requests, which we use to
+ * deterministically associate each response with the originating request.
+ */
+const assignSearchRequestsToSearchStrategies = searchRequests => {
+  const searchStrategiesWithRequests = [];
+  const searchStrategyById = {};
+
+  searchRequests.forEach(searchRequest => {
+    const matchingSearchStrategy = searchStrategies.find(searchStrategy => searchStrategy.isValidForSearchRequest(searchRequest));
+    const { id } = matchingSearchStrategy;
```

Comment:
getting an error for a rollup index pattern in Discover, as it doesn't match any registered search strategies:

<img width="972" alt="screen shot 2018-07-06 at 3 56 25 pm" src="https://user-images.githubusercontent.com/1965714/42403322-36e8e8ac-8135-11e8-821e-f731c46360fc.png">

maybe for queries with no matching search strategies, we set id to `default`?